wallet/src/main.rs

```diff
@@ -1,119 +1,126 @@
-
-    const SHAWN_PHRASE: &str = "news slush supreme milk chapter athlete soap sausage put clutch what kitten";
-    let (shawn_pair, _) = Pair::from_phrase(SHAWN_PHRASE, None)?;
-
-    println!("Seed is:: {}", seed.as_str());
-    let (provided_pair, _) = Pair::from_phrase(seed.as_str(), None)?;
-
-    // Genesis Coin Reference
-    let shawn_coin_ref = OutputRef {
-        tx_hash: H256::zero(),
-        index: 0u32,
-    };
-
-    // Construct a simple Transaction to spend Shawn genesis coin
-    let mut transaction = Transaction {
-        inputs: vec![
-            Input {
-                output_ref: shawn_coin_ref,
-                witness: vec![], // We will sign the total transaction so this should be empty
-            }
-        ],
-        outputs: vec![
-            Output {
-                payload: Coin::new(amount).into(),
-                redeemer: OuterRedeemer::SigCheck(SigCheck { owner_pubkey: provided_pair.public().into() }),
-            }
-        ],
-        verifier: OuterVerifier::Money(MoneyVerifier::Spend),
-    };
-
-    let signature = shawn_pair.sign(&transaction.encode());
-    transaction.inputs[0].witness = signature.encode();
-
-    // Calculate the OutputRef which also serves as the storage location
-    // In order to retrieve later
-    let new_coin_ref = OutputRef {
-        tx_hash: <BlakeTwo256 as Hash>::hash_of(&transaction.encode()),
-        index: 0,
-    };
-
-    // Send the transaction
-    let genesis_spend_hex = hex::encode(transaction.encode());
-    let params = rpc_params![genesis_spend_hex];
-    let genesis_spend_response: Result<String, _> =
-        client.request("author_submitExtrinsic", params).await;
-    println!(
-        "Node's response to genesis_spend transaction: {:?}",
-        genesis_spend_response
-    );
-
-    // Wait a few seconds to make sure a block has been authored.
-    sleep(Duration::from_secs(3));
-
-    // Retrieve new coins from storage
-    let (pubkey, new_coin_from_storage) = get_coin_from_storage(&new_coin_ref, &client).await?;
-    println!("Retrieved the new coin from storage {:?} with owner: {:?}", new_coin_from_storage, sp_core::hexdisplay::HexDisplay::from(&pubkey.encode()));
-    Ok(())
+    let client = HttpClientBuilder::default().build(cli.endpoint)?;
+
+    // Dispatch to proper subcommand
+    match cli.command {
+        Command::AmoebaDemo => amoeba::amoeba_demo(&client).await,
+        Command::VerifyCoin { ref_string } => {
+            let output_ref = OutputRef::decode(&mut &hex::decode(ref_string)?[..])?;
+            money::print_coin_from_storage(&output_ref, &client).await
+        }
+        Command::SpendCoins(args) => money::spend_coins(&client, args).await,
+    }
 }
 
-async fn get_coin_from_storage(
+/// Fetch an output from chain storage given an OutputRef
+pub async fn fetch_storage<R: Redeemer>(
     output_ref: &OutputRef,
     client: &HttpClient,
-) -> anyhow::Result<(H256, Coin)> {
+) -> anyhow::Result<Output<R>> {
```

Comment:
This is more of what I was going for initially but wasnt running short on time I like this instead